server/market/orderrouter.go

```diff
@@ -0,0 +1,556 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package market
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/decred/dcrdex/server/account"
+	"github.com/decred/dcrdex/server/asset"
+	"github.com/decred/dcrdex/server/comms/msgjson"
+	dex "github.com/decred/dcrdex/server/market/types"
+	"github.com/decred/dcrdex/server/matcher"
+	"github.com/decred/dcrdex/server/order"
+)
+
+const maxClockOffset = 10 // seconds
+
+// The AuthManager handles client-related actions, including authorization and
+// communications.
+type AuthManager interface {
+	Route(route string, handler func(account.AccountID, *msgjson.Message) *msgjson.Error)
+	Auth(user account.AccountID, msg, sig []byte) error
+	Sign(...msgjson.Signable)
+	Send(account.AccountID, *msgjson.Message)
+}
+
+// MarketTunnel is a connection to a market and information about existing
+// swaps.
+type MarketTunnel interface {
+	AddEpoch(order.Order) error
```

Comment:
Guessing `MarketTunnel.AddEpoch` would send the order to the market via a channel as I've prototyped.  Is that the general idea of `AddEpoch`?  I have to digest the other functions and how best to interface with market though.

Reply:
The Market could create a MarketTunnel and provide that...

Reply:
Related discussion at https://github.com/decred/dcrdex/pull/66/files#r339822578